### ISSUE TITLE
Link LS bandwidth values

### ISSUE THREAD

**Issue by slinkish** [sbezverk/gobmp#213]
I discovered that I was getting nonsensical values in the 'max_link_bw'. Upon investigation my 10G and beyond link BW values were overflowing the uint32 type for this value set at bps. I have updated my local code to set this at mbps inside the uint32 type. This works for myself but could invalidate use case for anyone still running T1/E1 very small link speeds. OpenBMP uses kbps in its object specification but with 400G in production and 800G specs out and about I fear the 4.2 tbps limit of kbps will be overflowed sooner than later. Open to talking about modifying these attributes to more exotic types (jsons double float or even stringify) but again wanted to see where inertia for existing data is at.

**Comment by sbezverk** [comment]
@slinkish Please check RFC 5305, it defines the encoding of Link's Maximum Bandwidth,
```
[3.4](https://www.rfc-editor.org/rfc/rfc5305#section-3.4).  Sub-TLV 9: Maximum Link Bandwidth

   This sub-TLV contains the maximum bandwidth that can be used on this
   link in this direction (from the system originating the LSP to its
   neighbors).  This is useful for traffic engineering.

   The maximum link bandwidth is encoded in 32 bits in IEEE floating
   point format.  The units are bytes (not bits!) per second.

   This sub-TLV is optional.  This sub-TLV SHOULD appear once at most in
   each extended IS reachability TLV.
   ```
   
`gobmp` follows this encoding.

**Comment by slinkish** [comment]
@sbezverk This is not about float conversion in gobmp:
https://github.com/sbezverk/gobmp/blob/b66eebae02f0693c70cdd6bccf4054f6815f3dcb/pkg/bgpls/bgpls-nlri.go#L411

These functions are commented that they are converting the Bytes to Bits and them returning the value as a uint32 type. uint32 can only store a max integer of roughly 4.2 billion, in bps this 4.2 Gbps

```
// GetMaxLinkBandwidth returns value of Maximum Link Bandwidth in bps
func (ls *NLRI) GetMaxLinkBandwidth() uint32 {
	for _, tlv := range ls.LS {
		if tlv.Type != 1089 {
			continue
		}
		return uint32(math.Float32frombits(binary.BigEndian.Uint32(tlv.Value))) * 8
```
Per the golang spec: https://go.dev/ref/spec#Conversions
```
In all non-constant conversions involving floating-point or complex values, if the result type cannot represent the value the conversion succeeds but the result value is implementation-dependent.
```
The result is when I am trying to use link bandwidths in the 100s of Gbps these values cannot be represented as uint32 types in bps and per the golang spec the result of return uint32(large_bw) is undefined. 

If gobmp is trying to pass the data as close to the RFC as possible then returning `binary.BigEndian.Uint32(tlv.Value)` would be sufficient. It is when the gobmp functions read the data as a float32 and then cast that value back as a uint32 that we get the undefined bits returned.

**Comment by sbezverk** [comment]
@brmcdoug need your input as any way to solve it, might cause some impact to already existing applications. The way I see it, it can be solved by either:
- returning tlv 1089 as is and let the client app to deal with decoding into bits per second; will break current client application logic if they use it;
- usign different type of variable for MaxLinkBandwidth (instead of uint32 we could use uint64 for the same field which will give pretty much unlimited space for growth; client applications will need to be recompiled with new definitions;)
- introducing a new field, MaxLinkBandwidthKbps as uint64, depreciate the old one MaxLinkBandwidth, just leave it there not to break any existing applications but stop populating it.

Appreciate your thoughts...

**Comment by brmcdoug** [comment]
Hi Serguei, personally I like the third option (MaxLinkBandwidthKbps as uint64, depreciate the old one MaxLinkBandwidth) best.

**Comment by slinkish** [comment]
There three values similar RFC types and same logic in gobmp:
```
	MaxLinkBW             uint32                        `json:"max_link_bw,omitempty"`
	MaxResvBW             uint32                        `json:"max_resv_bw,omitempty"`
	UnResvBW              []uint32                      `json:"unresv_bw,omitempty"`
```
I could potentially draft a PR in a couple of days for all three. I would need clarification on the handling of the old value, with the json tags as they are in the types would assume MaxLinkBW would get omitted once depopulated. Would you just want to supply the MaxLinkBWKbps value casted a uint32 in bps or have it zero'd?
Serguei, let me know what your thinking here, thanks for the feedback Bruce!

**Comment by sbezverk** [comment]
Ok, I do not have a strong opinion about proposed 3 options, so since you guys are users of gobmp and prefer 3rd option, then let's go with it.
I think let's keep old variables as 0, please add deprecation comment in the code as well as start Deprecated section in Readme, so people would be aware of these changes. Looking forward to your PR.

**Comment by sbezverk** [comment]
Closed with #214
